commands/poll.js

```diff
@@ -343,7 +379,12 @@ module.exports = {
       });
 
       setTimeout(async () => {
-        if (!pollEnded) {
+        clearInterval(checkPollExpiryTimer);
+
+        const checkPollMessage = await interaction.channel.messages.fetch(`${pollMessage.id}`);
+
+        if (checkPollMessage && !pollEnded) {
+
```

Comment:
Major: I understand that it may be possible that the server admin could delete the poll if he/she sees it as unnecessary rather than waiting for the author of the poll to end it. 

However, this block of code would actually stop the bot if the message is indeed deleted since the Discord API would throw an `Unknown Message` error (code 10008 to be exact).

Instead of doing this, we should use a try-catch block here. If the error is caught, just `return` and it shouldn't proceed with the logic below. From here, we probably won't need to use `checkPollMessage`.

You can probably do whatever you want with the `try` block though. It's up to you how you'd handle this as long as the error would be caught and the flow would stay the same.